tasks/farm5/ShortReadAlignmentTasks.wdl

```diff
@@ -3,6 +3,102 @@ version 1.0
 import "../../structs/farm5/RunTimeSettings.wdl"
 import "../../structs/ReferenceSequence.wdl"
 
+task SplitUpInputFile {
+  input {
+    File input_file
+    String sample_id
+    RunTimeSettings runTimeSettings
+  }
+
+  String lanelet_file_prefix = "lanelet_"
+
+  command {
+    mkdir lanelet_temp
+    cd lanelet_temp
+
+    # TODO - verify that file starts with right header!
+
+    # splits list of mappings into single files.  One line each.
+    grep '^~{sample_id}\t' ~{input_file} | split -l 1 - ~{lanelet_file_prefix}
+  }
+
+  runtime {
+    singularity: runTimeSettings.lftp_singularity_image
+    memory: 3000
+    cpu: "1"
+    lsf_group: select_first([runTimeSettings.lsf_group, "malaria-dk"])
+    lsf_queue: select_first([runTimeSettings.lsf_queue, "normal"])
+  }
+
+  output {
+      Array[File] lanelet_files = glob("lanelet_temp/~{lanelet_file_prefix}*")
+  }
+}
+
+task ReadLaneletFile {
+
+  input {
+    File lanelet_file
+    RunTimeSettings runTimeSettings
+  }
+
+  Array[Array[String]] tmp = read_tsv(lanelet_file)
```

Comment:
The task is not needed here since this can be done in wdl within the scatter.

Reply:
Good suggestion!